pkg/auth/round_trippers.go

```diff
@@ -0,0 +1,69 @@
+/*
+
+ Copyright 2021 The KubeVela Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package auth
+
+import (
+	"net/http"
+
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/client-go/transport"
+
+	"github.com/oam-dev/kubevela/pkg/multicluster"
+	oamutil "github.com/oam-dev/kubevela/pkg/oam/util"
+	"github.com/oam-dev/kubevela/pkg/utils"
+)
+
+var _ utilnet.RoundTripperWrapper = &impersonatingRoundTripper{}
+
+type impersonatingRoundTripper struct {
+	rt http.RoundTripper
+}
+
+// NewImpersonatingRoundTripper will add an ImpersonateUser header to a request
+// if the context has a specific user whom to act-as.
+func NewImpersonatingRoundTripper(rt http.RoundTripper) http.RoundTripper {
+	return &impersonatingRoundTripper{
+		rt: rt,
+	}
+}
+
+func (rt *impersonatingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	ctx := req.Context()
+
+	// Skip impersonation on non-local cluster requests
+	if !multicluster.IsInLocalCluster(ctx) {
```

Comment:
I think there is no necessary limit, when the user config the ServiceAccount, we can assume that it already exists in the target cluster.

Reply:
Is it okay to assume that both Namespace and ServiceAccount exist in the target cluster? (I was not sure about the existence of the Application namespace and logical equality of the namespace. For example, application namespace `app-ns` can be used in the target cluster for the other purpose)

Reply:
It's not safe for now, if we want to free this limit, we should have mechanism to keep it safe.

Reply:
@devholic Currently the namespace of the target cluster may be inconsistent with the app.

Reply:
@barnettZQG If so, keeping this limit seems better before implementing OCM Executor-like approaches. (https://github.com/oam-dev/kubevela/issues/3432#issuecomment-1066264448)